Update index.md (#4540)
mochajs/mocha: docs/index.md

This feature does re-run a failed test and its corresponding `beforeEach/afterEach` hooks, but not `before/after` hooks. `this.retries()` has no effect on failing hooks.

> **NOTE**: The following example was written using Selenium Webdriver (which [overwrites global Mocha hooks][selenium-webdriver-testing] for `Promise` chain).

Reviewer: ```suggestion
**NOTE**: The following example was written using the Selenium Webdriver (which [overwrites global Mocha hooks][selenium-webdriver-testing] for the `Promise` chain).
```

> _Updated in v4.0.0._

> **TL;DR:** If your tests hang after an upgrade to Mocha v4.0.0 or newer, use `--exit` for a quick (though not necessarily recommended) fix.

Reviewer: ```suggestion
**TL;DR:** If your tests are hanging after an upgrade to Mocha v4.0.0 or newer, use `--exit` for a quick (though not necessarily recommended) fix.
```

> _New in Node.js v12.12.0_

If the [`--enable-source-maps`](https://nodejs.org/dist/latest-v12.x/docs/api/cli.html#cli_enable_source_maps) flag

> is passed to Mocha, source maps will be collected and used to provide accurate stack traces for transpiled code:

Reviewer: isn't it supposed to be lowercase?

To migrate your tests using root hooks to a root hook plugin:

1. Find your root hooks (hooks defined _outside_ of a suite&mdash;usually `describe()` callback).

> 1. Create a new file (e.g., `test/hooks.js`).

Reviewer: ```suggestion
1. Create a new file (for example, `test/hooks.js`).
```

To use it, load this file when running Mocha via `mocha --require fixtures.cjs` (or whatever you have named the file).

> > _Remember:_ You can define "requires" in a [configuration file](#configuring-mocha-nodejs).

Reviewer: ```suggestion
> _Reminder:_ You can define `requires` in a [configuration file](#configuring-mocha-nodejs).
```

When Mocha itself throws exception, the associated `Error` will have a `code` property. Where applicable, consumers should check the `code` property instead of string-matching against the `message` property. The following table describes these error codes:

> | Code                               | Description                                                  |
> | ---------------------------------- | ------------------------------------------------------------ |
> | `ERR_MOCHA_INVALID_ARG_TYPE`       | wrong type was passed for a given argument                   |
> | `ERR_MOCHA_INVALID_ARG_VALUE`      | invalid or unsupported value was passed for a given argument |
> | `ERR_MOCHA_INVALID_EXCEPTION`      | a falsy or otherwise underspecified exception was thrown     |
> | `ERR_MOCHA_INVALID_INTERFACE`      | interface specified in options not found                     |
> | `ERR_MOCHA_INVALID_REPORTER`       | reporter specified in options not found                      |
> | `ERR_MOCHA_NO_FILES_MATCH_PATTERN` | test file(s) could not be found                              |
> | `ERR_MOCHA_UNSUPPORTED`            | requested behavior, option, or parameter is unsupported      |

Reviewer: ```suggestion
| Code                               | Description                                                  |
| ---------------------------------- | ------------------------------------------------------------ |
| `ERR_MOCHA_INVALID_ARG_TYPE`       | Wrong type was passed for a given argument                   |
| `ERR_MOCHA_INVALID_ARG_VALUE`      | Invalid or unsupported value was passed for a given argument |
| `ERR_MOCHA_INVALID_EXCEPTION`      | A falsy or otherwise underspecified exception was thrown     |
| `ERR_MOCHA_INVALID_INTERFACE`      | Interface specified in options not found                     |
| `ERR_MOCHA_INVALID_REPORTER`       | Reporter specified in options not found                      |
| `ERR_MOCHA_NO_FILES_MATCH_PATTERN` | Test file(s) could not be found                              |
| `ERR_MOCHA_UNSUPPORTED`            | Requested behavior, option, or parameter is unsupported      |
```
